api/src/services/collections.ts

```diff
@@ -617,7 +620,24 @@ export class CollectionsService {
 						schema: this.schema,
 					});
 
-					await trx('directus_fields').delete().where('collection', '=', collectionKey);
+					const fieldItemsService = new ItemsService('directus_fields', {
+						knex: trx,
+						accountability: this.accountability,
+						schema: this.schema,
+					});
+
+					await fieldItemsService.deleteByQuery(
+						{
+							filter: {
+								collection: { _eq: collectionKey },
+							},
+						},
+						{
+							bypassEmitAction: (params) =>
+								opts?.bypassEmitAction ? opts.bypassEmitAction(params) : nestedActionEvents.push(params),
+						},
+					);
+
```

Comment:
What's the benefit of using the the ItemsService here instead of deleting the proper `directus_fields` directly?

Reply:
Ah, I'm guessing this is related to
> We now emit the fields.delete event on collections.delete

right?
In that case this is good to go, if I'm incorrect though, please let me know.

Reply:
Yes, this is to ensure we emit the `fields.delete` event when the records are removed.